Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TriangleFEMForceField.h

```diff
@@ -59,23 +59,15 @@ class TriangleFEMForceField : public core::behavior::ForceField<DataTypes>
     SOFA_CLASS(SOFA_TEMPLATE(TriangleFEMForceField, DataTypes), SOFA_TEMPLATE(core::behavior::ForceField, DataTypes));
 
     typedef core::behavior::ForceField<DataTypes> Inherited;
-    typedef typename DataTypes::VecCoord VecCoord;
-    typedef typename DataTypes::VecDeriv VecDeriv;
-    typedef typename DataTypes::Coord    Coord   ;
-    typedef typename DataTypes::Deriv    Deriv   ;
-    typedef typename Coord::value_type   Real    ;
 
-    typedef core::objectmodel::Data<VecCoord> DataVecCoord;
-    typedef core::objectmodel::Data<VecDeriv> DataVecDeriv;
-
-    typedef type::Vec<6, Real> Displacement;								///< the displacement vector
-    typedef type::Mat<3, 3, Real> MaterialStiffness;						///< the matrix of material stiffness
+    typedef type::Vec<6, Real_t<DataTypes>> Displacement;								///< the displacement vector
+    typedef type::Mat<3, 3, Real_t<DataTypes>> MaterialStiffness;						///< the matrix of material stiffness
     typedef sofa::type::vector<MaterialStiffness> VecMaterialStiffness;    ///< a vector of material stiffness matrices
-    typedef type::Mat<6, 3, Real> StrainDisplacement;						///< the strain-displacement matrix (the transpose, actually)
+    typedef type::Mat<6, 3, Real_t<DataTypes>> StrainDisplacement;						///< the strain-displacement matrix (the transpose, actually)
```

Comment:
Mixing real precision in our codebase has the only consequence of making it very harde to track the preservation of a given level of precision (i.e, there is no undesired/un-expected loss). 
In previously merged PR related to refactoring/removal of alias declaration the idea was to simply use SReal. So we are compiling SOFA either in double or in float and not a giant mic-mac.